proto/config/config.proto

```diff
@@ -0,0 +1,95 @@
+syntax = "proto3";
+
+package linkerd2.config;
+
+option go_package = "github.com/linkerd/linkerd2/controller/gen/config";
+
+// ConfigStore provides an abstraction over the underlying configuration storge.
+service ConfigStore {
+  // Return the JSON-serialized global config.
+  rpc GetGlobalConfig(GetParam) returns (GlobalConfig) {}
+
+  // Return the JSON-serialized proxy config.
+  rpc GetProxyConfig(GetParam) returns (ProxyConfig) {}
+
+  // Set the global config. It overwrites the existing value.
+  rpc SetGlobalConfig(GlobalConfig) returns (Result) {}
+
+  // Set the proxy config. It overwrites the existing value.
+  rpc SetProxyConfig(ProxyConfig) returns (Result) {}
+}
```

Comment:
Yes, the intent is to provide an API to insulate all the consumers from whatever goes on at the k8s API and config storage level.

Reply:
Let's think about this a little... is the go-generated Service code going to be the best API for interacting with the config?

When will the `Set` calls _ever_ make sense in this?

I think we care about protobuf _mostly_ for having reliable serialization; but we don't have any need to serialize requests/repsonses over a network; so this service abstraction seems unnecessary to me.

At the very least, it seems like something we can add separately from identifying the core data types, which is most important right now.

Reply:
Perhaps it should be a goal for the "API" in our applications to _hide_ the actual protobuf structs (which may change over time) and instead expose a higher-level interface that can be satisfied by reading the protobuf types.

For instance, having `config.IsCNIEnabled()` would be nice, so that I don't have to worry about whether that config is stored in global or inject config... etc.

Reply:
I do think having some form of "API" over the configs is better than each container having to mount the config map, performs its own JSON serialization/deserialization and dealing with the protobuf types directly.

And yes, it's does make more sense to add something like this to the shared library, instead of over the network. I will remove this endpoint.

Also, since install, inject and auto-inject only work with JSON blobs, we can introduce individual accessors like `config.IsCNIEnabled()` later when the needs arise, to limit this PR to only what is currently needed.